.github/workflows/firebase.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy Backend to Firebase Functions
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    
+    - name: Setup Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: 18
+        
+    - name: Install Backend Dependencies
+      working-directory: apps/backend/functions
+      run: npm install
+
+    - name: Install Firebase Tools
+      working-directory: apps/backend
+      run: npm install -g firebase-tools
+    
+    - name: Build Backend
+      working-directory: apps/backend/functions
+      run: npm run build  
+    
+    - name: Deploy to Firebase Functions
+      working-directory: apps/backend  
+      run: |
+        firebase login:ci --no-localhost --token "$FIREBASE_TOKEN"  # Login to Firebase using a CI token
```

Comment:
Where should I add FIREBASE_TOKEN  in the repo??

Reply:
add in the repository secret
@hkirat